central/role/datastore/datastore_impl.go

```diff
@@ -287,11 +304,32 @@ func (ds *dataStoreImpl) GetAllPermissionSets(ctx context.Context) ([]*storage.P
 	return permissionSets, nil
 }
 
+func (ds *dataStoreImpl) GetPermissionSetsFiltered(ctx context.Context,
+	filter func(permissionSet *storage.PermissionSet) bool) ([]*storage.PermissionSet, error) {
```

Comment:
I think in the ideal world the filter should be a query that can be transformed into SQL `WHERE` clause. However, looking at how it is currently implemented for policies this would definitely be overkill to reimplement here.

Reply:
Yeah I was thinking the same thing. For that to be supported I suppose we have to a) expose search functionality for the datastore b) make the origin field searchable.
Maybe we can use the query select framework for that.

I created https://issues.redhat.com/browse/ROX-16071 to cover that one.